### PR TITLE
[Fix]Support HGT mini-batch

### PR DESCRIPTION
**Pull request by dddg617**
## Description

Support mini-batch in dgl.nn.pytorch.conv.hgtconv. Related issue is https://github.com/dmlc/dgl/issues/4482 and https://github.com/dmlc/dgl/issues/4643. We separate source and target node IDs and types. This will not affect the full batch training and I have tested mini-batch in OpenHGNN.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [X] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [X] Code is well-documented
- [X] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

## Changes

- [X] dgl.nn.pytorch.conv.hgtconv.

### CODE REVIEW

**Comment by isratnisa** [comment]
Hi! Thanks for the code. Can you please compare the performance number with the old version? Also, please add numbers for minibatch version so we know if there is any issue we need to resolve.

**Comment by dddg617** [comment]
After testing, we have three tables. And we use the same parameters when testing.

Hyperparameters: num_heads = 4, in_dim = 256, num_layers = 2, dropout = 0.2, clip = 1.0, max_lr = 1e-3.
Dataset: ACM.mat.
Graph(num_nodes={'author': 17431, 'paper': 12499, 'subject': 73},
      num_edges={('author', 'writing', 'paper'): 37055, ('paper', 'cited', 'paper'): 30789, ('paper', 'citing', 'paper'): 30789, ('paper', 'is-about', 'subject'): 12499, ('paper', 'written-by', 'author'): 37055, ('subject', 'has', 'paper'): 12499}
)
Evaluation metric: accuracy

Original HGTConv (full-batch train):
|    val     |    test     |
| ---------- | ----------- |
| 0.484±0.05 | 0.425±0.004 |

New HGTConv (full-batch train):
|     val     |    test     |
| ----------- | ----------- |
| 0.468±0.057 | 0.426±0.010 |

New HGTConv (mini-batch train) (batch_size = 64):
|    val     |    test     |
| ---------- | ----------- |
| 0.442±0.04 | 0.411±0.021 |

**Comment by isratnisa** [comment]
Thanks for the tables. Is this the expected accuracy? Can you please double check if the accuracy on ACM and DBLP dataset matches with the [official example](https://github.com/UCLA-DM/pyHGT)?
Also, please resolve the conflict. Thanks.

**Comment by dddg617** [comment]
OK, but the offical example do not have ACM and DBLP datasets. So do we need to test this on the dataset OAG?

**Comment by dddg617** [comment]
OK, but the offical example do not have ACM and DBLP datasets. So do we need to test this on the dataset OAG?

**Comment by isratnisa** [comment]
OAG should be fine. I found a [PyG example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/hgt_dblp.py) on dblp.

**Comment by isratnisa** [comment]
Also, please add the training code where you are using `HGTConv`.

**Comment by dddg617** [comment]
OK, I will do this weekend.

**Review by jermainewang — CHANGES_REQUESTED** [review]
Waiting for @dddg617 to resolve the conflict and add unit tests. Examples that use this new feature can be in another PR. cc @isratnisa

**Comment by jermainewang** [comment]
@dgl-bot

**Review by jermainewang — APPROVED** [review]

**Comment by jermainewang** [comment]
@dddg617 Thanks for the contribution!
